src/global/project/manifest.rs

```diff
@@ -431,40 +431,52 @@ impl Manifest {
 #[derive(Debug, Clone, Serialize, Deserialize, Hash, PartialEq, Eq)]
 pub struct Mapping {
     exposed_name: ExposedName,
-    executable_name: String,
+    executable_relname: String,
 }
 
 impl Mapping {
-    pub fn new(exposed_name: ExposedName, executable_name: String) -> Self {
+    pub fn new(exposed_name: ExposedName, executable_relname: String) -> Self {
         Self {
             exposed_name,
-            executable_name,
+            executable_relname: Path::new(&executable_relname)
+                .with_extension("")
+                .to_str()
+                .unwrap_or(&executable_relname)
+                .to_string(),
```

Comment:
with extention doesn't work for `python3.12` as that will create `python3`. 

You can use `pixi_utils::executable_from_path` here aswell.

Reply:
I've just changed new method, I've used strip_executable_extension (because executable_from_path drop nested directory)